### PR TITLE
chore: use `clap` to have command line arguments

### PR DESCRIPTION
**Pull request by drupol**
This PR:

- Let `clap` handle the command line arguments and options

image

I'm pretty new with rust, please let me know if I'm doing something wrong, I'm here to learn :)

### CODE REVIEW

File: src/parser/error.rs [hunk 1]
```diff
@@ -108,12 +108,10 @@ impl ParseError {
     pub fn report<'a>(
         &self,
         source: &'a str,
-        origin: Option<&'a str>,
+        origin: &std::string::String,
         colored: bool,
         ascii: bool,
     ) -> std::io::Result<String> {
-        let origin = origin.unwrap_or("input");
-
```
Comment (azjezz) [inline src/parser/error.rs:116]: please revert this change, we want `Option<&str>`, even tho we now just unwrap it to `input`, there's plans to change that into something else in the future.

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,3 @@
+/.direnv
```
Comment (azjezz) [inline .gitignore:1]: please have this in global gitignore instead of here.

File: .envrc [hunk 3]
```diff
@@ -0,0 +1 @@
+use flake github:loophp/rust-shell
```
Comment (azjezz) [inline .envrc:1]: same here.

File: src/parser/error.rs (hunk 1 above)
Comment (drupol) [reply src/parser/error.rs]: Fixed

File: src/main.rs [hunk 4]
```diff
@@ -1,23 +1,25 @@
...
-        std::process::exit(0);
-    }
+    let args = Arguments::parse();
 
-    // get file from args or print error and exit
-    let file = args.get(1).unwrap();
-    let silent = args.contains(&String::from("--silent"));
-    let print_json = args.contains(&String::from("--json"));
-    let contents = std::fs::read_to_string(file)?;
+    let file = args.file.unwrap();
```
Comment (azjezz) [inline src/main.rs:19]: wouldn't `php-parser-rs --json` panic here?

Comment (drupol) [reply src/main.rs:19]: nope

image

Comment (azjezz) [reply src/main.rs:19]: I think it does, let's make `file` argument `String`, and pass `Some(&file)` to `report()`.

Comment (drupol) [reply src/main.rs:19]: oops, you're right !

Going to fix this.

Comment (drupol) [reply src/main.rs:19]: I'm kind of blocked here. How should we deal with that? What do you advise?

image

Comment (drupol) [reply src/main.rs:19]: Got it!

Comment (drupol) [reply src/main.rs]: image

Comment (azjezz) [reply src/main.rs]: well, you are on the right path for learning Rust, it's mostly trial and error until you figure it out 😋

**Comment by azjezz** [comment]
is seems there's few formating issues, you can run `just fix` to fix them.

**Comment by azjezz** [comment]
Thanks @drupol :D

**Comment by drupol** [comment]
Let me know if there are some easy task I could tackle!

**Pull request merged**